languages/ocaml.scm

```diff
@@ -784,42 +783,49 @@
   ] @append_spaced_softline @prepend_spaced_softline
 )
 
-; Put a semicolon delimiter after field declarations, unless they already have
-; one, in which case we do nothing.
-; The semicolon always comes right after the declaration of the new field,
-; before any comment.
+; Put a semicolon delimiter after field declarations and potential ppx attributes,
+; unless they already have one, in which case we do nothing.
+; The semicolon always comes right after the declaration of the new field
+; and attributes, before any comment.
 ; Hence, if there is a comment between the field declaration and the associated ";",
 ; the semicolon is moved before the comment.
 ;
 ; type t =
-;   { mutable position : int (* End-of-line comment *);
+;   { mutable position : int [@default 0] (* End-of-line comment *);
 ;   ...
 ;
 ; is turned into
 ;
 ; type t =
 ;   {
-;     mutable position : int; (* End-of-line comment *)
+;     mutable position : int [@default 0]; (* End-of-line comment *)
 ;     ...
 ;
 (
   (field_declaration) @append_delimiter
   .
   [
     ";"
-    (comment)
+    (attribute)
   ]* @do_nothing
   (#delimiter! ";")
 )
 (
-  (field_declaration) @append_delimiter
+  (field_declaration)
+  .
+  (attribute)+ @append_delimiter
```

Comment:
Doesn't this break the case where there is no attribute?

Reply:
This is handled by the case above:
```scheme
(
  (field_declaration) @append_delimiter
  .
  [
    ";"
    (comment)
    (attribute)
  ]* @do_nothing
  (#delimiter! ";")
)
```

Reply:
What about the comment? I am only seeing:

```
(
  (field_declaration) @append_delimiter
  .
  [
    ";"
    (attribute)
  ]* @do_nothing
  (#delimiter! ";")
)
```

Reply:
You're right, I copy/pasted the diff too quickly.
The query in your message takes care of the case when there is no attribute.
Do you think there is an example that would get mis-formatted with the new queries?

Reply:
I was worried that `(comment)` on line 810 got lost, but that actually seems like the right thing to do.